src/c-writer.cc

```diff
@@ -1745,6 +1820,67 @@ void CWriter::WriteImports() {
   }
 }
 
+void CWriter::WriteTailCallWeakImports() {
+  for (const Import* import : unique_imports_) {
+    if (import->kind() != ExternalKind::Func) {
+      continue;
+    }
+    const Func& func = cast<FuncImport>(import)->func;
+    Write(Newline(), "/* handler for missing tail-call on import: '",
+          SanitizeForComment(import->module_name), "' '",
+          SanitizeForComment(import->field_name), "' */", Newline());
+    Write("#ifdef _MSC_VER", Newline(),
+          "#pragma comment(linker, \"/alternatename:",
+          TailCallExportName(import->module_name, import->field_name), "=",
+          kTailCallFallbackPrefix, module_prefix_, "_",
+          ExportName(import->module_name, import->field_name), "\")",
+          Newline());
+    WriteTailCallFuncDeclaration(
+        kTailCallFallbackPrefix + module_prefix_ + '_' +
+        ExportName(import->module_name, import->field_name));
+    Write(Newline(), "#else", Newline());
+    Write("__attribute__((weak)) ");
+    WriteTailCallFuncDeclaration(
+        TailCallExportName(import->module_name, import->field_name));
+    Write(Newline(), "#endif", Newline(), OpenBrace());
+    Write("next->fn = NULL;", Newline());
```

Comment:
Would it be readable/maintainable to define a macro in the internal header rather than try to stamp out the `if/else/endif` here?

Reply:
Done (or at least attempted -- doing the #pragma inside a #define was a little tricky).